daprdocs/content/en/developing-applications/building-blocks/state-management/howto-outbox.md

```diff
@@ -192,7 +345,128 @@ By setting the metadata item `"outboxProjection"`  to `"true"`, the first transa
 
 You can also override the [Dapr-generated CloudEvent fields]({{< ref "pubsub-cloudevents.md#dapr-generated-cloudevents-example" >}}) on the published outbox event with custom CloudEvent metadata.
 
-{{< tabs "Go SDK" HTTP >}}
+{{< tabs JavaScript ".NET" Java Go HTTP >}}
+
+{{% codetab %}}
+
+<!--javascript-->
+
+```javascript
+const { DaprClient } = require('dapr-client');
+
+async function executeStateTransaction() {
+    // Initialize Dapr client
+    const daprClient = new DaprClient();
+
+    // Define state operations
+    const ops = [];
+
+    const op1 = {
+        operationType: 'upsert',
+        request: {
+            key: 'key1',
+            value: Buffer.from('2'),
+            metadata: {
+                'id': 'unique-business-process-id',
+                'source': 'CustomersApp',
+                'type': 'CustomerCreated',
+                'subject': '123',
+                'my-custom-ce-field': 'abc'
+            }
+        }
+    };
+
+    ops.push(op1);
+
+    // Execute state transaction
+    const storeName = 'your-state-store-name';
+    const metadata = {};
+}
+
+executeStateTransaction();
+```
+{{% /codetab %}}
+
+{{% codetab %}}
+
+<!--csharp-->
+
+```csharp
+public class StateOperationExample
```

Comment:
Not sure where this code came from, however it should be like this

```
public class StateOperationExample
{
    public async Task ExecuteStateTransactionAsync()
    {
        var daprClient = new DaprClientBuilder().Build();

        // Define the value "2" as a string and serialize it to a byte array
        var value = "2";
        var valueBytes = JsonSerializer.SerializeToUtf8Bytes(value);

        // Define the first state operation to save the value "2" with metadata
       // Override Cloudevent metadata
        var metadata = new Dictionary<string, string>
        {
            { "cloudevent.id", "unique-business-process-id" },
            { "cloudevent.source", "CustomersApp" },
            { "cloudevent.type", "CustomerCreated" },
            { "cloudevent.subject", "123" },
            { "my-custom-ce-field", "abc" }
        };

        var op1 = new StateTransactionRequest(
            key: "key1",
            value: valueBytes,
            operationType: StateOperationType.Upsert,
            metadata: metadata
        );

        // Create the list of state operations
        var ops = new List<StateTransactionRequest> { op1 };

        // Execute the state transaction
        var storeName = "your-state-store-name";
        await daprClient.ExecuteStateTransactionAsync(storeName, ops);
        Console.WriteLine("State transaction executed.");
    }

    public static async Task Main(string[] args)
    {
        var example = new StateOperationExample();
        await example.ExecuteStateTransactionAsync();
    }
}
```

Reply:
it came from ChatGPT, like you suggested :)